hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -14,26 +14,36 @@
   param_list: [
     { name: "ByteOrder",
       desc: '''Byte order to use when transmitting or receiving data.  If ByteOrder = 0,
-               the IP uses a Big-Endian ordering for the bytes in TXDATA and RXDATA.
-               The most significant byte (MSB) of TXDATA is transmitted first, and
-               received data is placed in the MSB location of RXDATA.  If ByteOrder = 1,
+               the IP uses a Big-Endian ordering for the bytes in DATA.
+               The most significant byte (MSB) of DATA is transmitted first, and
+               received data is placed in the MSB location of DATA.  If ByteOrder = 1,
                a Little-Endian ordering is used for these registers, and the LSB of each
                gets priority for receiving and transmitting data.'''
-      type: "int",
+      type: "logic",
       default: "0"
```

Comment:
can we default this to 1 since our system is little endian? :D

Reply:
Oh.  Ha.  Right. Sure thing.

Though as a side note this is actually not entirely an obvious choice as flash devices are "Big-endian on the bus" meaning that for addresses they expect the MSB first.   However for the bulk of data transfers, I agree, you want data to be in the same order as it's laid out in CPU memory, so yeah, we want this to match Ibex.  So "Default 1" it is

Reply:
THIS.  What you said is EXACTLY what caused my initial confusion and many questions on the bit and byte ordering :)

Reply:
Question for DV planning?
Do we want to also prioritize ByteOrder == 1 test cases?  My guess is that there won't even be a DIF that uses the Big-Endian setting.
(FYI @tunghoang290780)

Reply:
just a note that since the parameter is listed in the hjson, there's no way for DV to change it anyways just because of the way our environment is setup.

Reply:
@martin-lueker @tjaychen the test case will prioritize the default values which are initialized in RTL reg. package. Those parameters are known at compile time thus DV can not dynamically re-configure them during run-time (at least I do not know how to do). As said, DV will focus on ByteOrder=1.